tests/components/sharkiq/test_shark_iq.py

```diff
@@ -0,0 +1,283 @@
+    assert state_json == target_json
+
+
+def _get_async_update(err=None):
+    async def _async_update(_) -> bool:
+        if err is not None:
+            raise err
+        return True
+
+    return _async_update
+
+
+@patch.object(AylaApi, "async_list_devices", new=_async_list_devices)
+async def test_updates(hass: HomeAssistant) -> None:
+    """Test the update coordinator update functions."""
+    ayla_api = get_ayla_api(TEST_USERNAME, TEST_PASSWORD)
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    mock_config = MagicMock(spec=ConfigEntry)
+    coordinator = SharkIqUpdateCoordinator(hass, mock_config, ayla_api, [shark_vac])
+
+    with patch.object(SharkIqVacuum, "async_update", new=_get_async_update()):
+        update_called = (
+            await coordinator._async_update_data()  # pylint: disable=protected-access
+        )
+    assert update_called
+
+    update_failed = False
+    with patch.object(
+        SharkIqVacuum, "async_update", new=_get_async_update(SharkIqAuthError)
+    ), patch.object(HomeAssistant, "async_create_task"), patch.object(
+        ConfigEntriesFlowManager, "async_init"
+    ):
+        try:
+            await coordinator._async_update_data()  # pylint: disable=protected-access
+        except UpdateFailed:
+            update_failed = True
+    assert update_failed
+
+
+async def test_coordinator_match(hass: HomeAssistant):
+    """Test that sharkiq-coordinator references work."""
+    ayla_api = get_ayla_api(TEST_PASSWORD, TEST_USERNAME)
+    shark_vac1 = _get_mock_shark_vac(ayla_api)
+    shark_vac2 = _get_mock_shark_vac(ayla_api)
+    shark_vac2._dsn = "FOOBAR!"  # pylint: disable=protected-access
+
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac1])
+
+    # The first should succeed, the second should fail
+    api1 = SharkVacuumEntity(shark_vac1, coordinator)
+    try:
+        _ = SharkVacuumEntity(shark_vac2, coordinator)
+    except RuntimeError:
+        api2_failed = True
+    else:
+        api2_failed = False
+    assert api2_failed
+
+    coordinator.last_update_success = True
+    coordinator._online_dsns = set()  # pylint: disable=protected-access
+    assert not api1.is_online
+    assert not api1.available
+
+    coordinator._online_dsns = {  # pylint: disable=protected-access
+        shark_vac1.serial_number
+    }
+    assert api1.is_online
+    assert api1.available
+
+    coordinator.last_update_success = False
+    assert not api1.available
+
+
+async def test_simple_properties(hass: HomeAssistant):
+    """Test that simple properties work as intended."""
+    ayla_api = get_ayla_api(TEST_PASSWORD, TEST_USERNAME)
+    shark_vac1 = _get_mock_shark_vac(ayla_api)
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac1])
+    entity = SharkVacuumEntity(shark_vac1, coordinator)
+
+    assert entity.unique_id == "AC000Wxxxxxxxxx"
+
+    assert entity.supported_features == (
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)